### PR TITLE
Added trucation mode to reorder for convert

### PR DESCRIPTION
**Pull request by steve-y**
### Details:
 - *Added trucation mode to reorder when it is transformed to convert operation*

### Tickets:
 - *92497*

### CODE REVIEW

File: src/plugins/intel_gpu/include/intel_gpu/primitives/reorder.hpp [hunk 1]
```diff
@@ -143,6 +146,8 @@ struct reorder : public primitive_base<reorder> {
     std::vector<float> subtract_per_feature;
     /// @brief Mode of mean execution
     reorder_mean_mode mean_mode;
+    /// @brief Convert truncation Mode
+    bool truncate;
```
Comment (yeonbok) [inline src/plugins/intel_gpu/include/intel_gpu/primitives/reorder.hpp:150]: Shouldnt' it have default value?

File: src/plugins/intel_gpu/src/kernel_selector/cl_kernels/reorder_data.cl [hunk 2]
```diff
@@ -117,7 +117,7 @@ KERNEL (reorder_data)(
 #else
     uint8 ov = RESHAPE_DIMS(INPUT0, OUTPUT, b, f, w, z, y, x);
     const uint input_idx  = FUNC_CALL(get_input_index)(b, f, w, z, y, x);
-    const uint output_idx = FUNC_CALL(get_output_index)(ov.s1,ov.s2,ov.s3,ov.s4,ov.s5,ov.s6);
+    const uint output_idx = FUNC_CALL(get_output_index)(ov[1],ov[2],ov[3],ov[4], ov[5], ov[6]);
```
Comment (yeonbok) [inline src/plugins/intel_gpu/src/kernel_selector/cl_kernels/reorder_data.cl:120]: Why this change is needed?

File: src/plugins/intel_gpu/src/kernel_selector/cl_kernels/reorder_data.cl [hunk 3]
```diff
@@ -130,7 +130,7 @@ KERNEL (reorder_data)(
     // TODO Add support for 6D mean
     MEAN_SUBTRACT_TYPE res = TO_MEAN_TYPE(input[input_idx]);
     uint8 msv = RESHAPE_DIMS(INPUT0, MEAN_SUBTRACT, b, f, w, z, y, x);
-    res = MEAN_OP(res, mean_subtract[GET_DATA_INDEX_SAFE(MEAN_SUBTRACT, msv.s1, msv.s2, /*msv.s3, msv.s4,*/ msv.s5, msv.s6)]);
+    res = MEAN_OP(res, mean_subtract[GET_DATA_INDEX_SAFE(MEAN_SUBTRACT, msv[1], msv[2], /*msv[3], msv[4],*/ msv[5], msv[6])]);
```
Comment (yeonbok) [inline src/plugins/intel_gpu/src/kernel_selector/cl_kernels/reorder_data.cl:133]: Here too. Why this should be modifed?

File: src/plugins/intel_gpu/src/kernel_selector/cl_kernels/reorder_data.cl [hunk 4]
```diff
@@ -139,34 +139,38 @@ KERNEL (reorder_data)(
...
-    output_idx = FUNC_CALL(get_output_index)(ov.s1, ov.s2, ov.s3, ov.s4, ov.s5, ov.s6);
+    output_idx = FUNC_CALL(get_output_index)(ov[1], ov[2], ov[3], ov[4], ov[5], ov[6]);
     output[output_idx] = ACTIVATION_FUNC_TYPED(OUTPUT_REORDER, TO_OUTPUT_REORDER_TYPE(colorRGBA.s3), NL_M, NL_N);
 #endif
 #elif OUTPUT_LAYOUT_IMAGE_2D_RGBA
     IMAGE_WRITE(output, (int2)(x, y), colorRGBA);
 #else
 #if INPUT0_IS_FP && !OUTPUT_IS_FP
+#if CONVERT_TRUNCATE
+    output[output_idx] = ACTIVATION_TYPED(OUTPUT_REORDER, TO_OUTPUT_REORDER_TYPE_SAT((OUTPUT_REORDER_TYPE)(long)res), ACTIVATION_PARAMS_TYPED);
```
Comment (yeonbok) [inline src/plugins/intel_gpu/src/kernel_selector/cl_kernels/reorder_data.cl:170]: Please add corresponding unittests.

Comment (steve-y) [reply src/plugins/intel_gpu/src/kernel_selector/cl_kernels/reorder_data.cl:170]: Fixed it as it was wrongly rebased.

Comment (steve-y) [reply src/plugins/intel_gpu/src/kernel_selector/cl_kernels/reorder_data.cl:170]: Added unittest reorder_gpu.basic_convert_uint8.

File: src/plugins/intel_gpu/src/kernel_selector/cl_kernels/reorder_data.cl (hunk 2 above)
Comment (steve-y) [reply src/plugins/intel_gpu/src/kernel_selector/cl_kernels/reorder_data.cl]: Fixed it as it was wrongly rebased.

File: src/plugins/intel_gpu/src/kernel_selector/cl_kernels/reorder_data.cl (hunk 3 above)
Comment (steve-y) [reply src/plugins/intel_gpu/src/kernel_selector/cl_kernels/reorder_data.cl]: Fixed it as it was wrongly rebased.

File: src/plugins/intel_gpu/include/intel_gpu/primitives/reorder.hpp (hunk 1 above)
Comment (yeonbok) [reply src/plugins/intel_gpu/include/intel_gpu/primitives/reorder.hpp:150]: yes because there are another constructor

Comment (steve-y) [reply src/plugins/intel_gpu/include/intel_gpu/primitives/reorder.hpp]: Updated as your comment, thanks

**Comment by yeonbok** [comment]
Did you check affected networks?

**Comment by steve-y** [comment]
> Did you check affected networks?

Yes, text-spotting-0001-recognizer-decoder and text-spotting-0002-recognizer-decoder show accuracy regression, so I'm checking on them.

**Comment by steve-y** [comment]
> > Did you check affected networks?
> 
> Yes, text-spotting-0001-recognizer-decoder and text-spotting-0002-recognizer-decoder show accuracy regression, so I'm checking on them.

text-spotting-0001-recognizer-decoder and text-spotting-0002-recognizer-decoder were found out that they have no accuracy regression. And all affected network were checked.

**Review by yeonbok — DISMISSED** [review]

File: src/plugins/intel_gpu/src/plugin/ops/convert.cpp [hunk 5]
```diff
@@ -41,7 +41,9 @@ static void CreateConvertOp(Program& p, const std::shared_ptr<ngraph::op::v0::Co
                                       cldnn::format::any,
                                       outDataType,
                                       std::vector<float>(),
-                                      cldnn::reorder_mean_mode::subtract);
+                                      cldnn::reorder_mean_mode::subtract,
+                                      cldnn::padding(),
+                                      true);
```
Comment (yeonbok) [inline src/plugins/intel_gpu/src/plugin/ops/convert.cpp:46]: Should this be always truncate?Can we get any condition from ngraph when to make it saturate or truncate?

**Review by yeonbok — APPROVED** [review]

File: src/plugins/intel_gpu/src/plugin/ops/convert.cpp (hunk 5 above)
Comment (yeonbok) [reply src/plugins/intel_gpu/src/plugin/ops/convert.cpp:46]: Aligned offline. All "Cast" operations are to be truncated.
